prog/kubernetes/kubernetes_cluster_nexus.rb

```diff
@@ -16,14 +16,25 @@ def self.assemble(name:, version:, private_subnet_id:, project_id:, location:, c
       Validation.validate_kubernetes_name(name)
       Validation.validate_kubernetes_cp_node_count(cp_node_count)
 
-      # TODO: Validate subnet location if given
-      # TODO: Validate subnet size if given
-      # TODO: Create subnet if not given
+      subnet = if private_subnet_id && private_subnet_id != ""
+        fail "Given subnet is not available in the given project" unless project.private_subnets.any? { |ps| ps.id == private_subnet_id }
+        PrivateSubnet[private_subnet_id]
+      else
+        subnet_name = name + "-k8s-subnet"
```

Comment:
Can it exceed the length limit of a private subnet?

Reply:
As far as I can see, `SubnetNexus.assemble` uses `validate_name`, which accepts names up to 63 characters. Kubernetes accepts names up to 40 characters.

Reply:
I didn't know there was a 40 character limit for the Kubernetes cluster name